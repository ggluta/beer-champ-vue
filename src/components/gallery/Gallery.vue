<template>
  <v-container id="gallery" fluid>
    <v-layout row wrap>
      <v-flex lg3 md4 sm6 xs12 v-for="beer in this.filteredAppData" :key="beer.title">
        <v-card class="ma-5 pa-5">
          <v-img contain height="300px" :src="getBeerImage(beer)">
          </v-img>
          <v-card-title class="text-md-h5 text-break text-ellipsis">
            {{ getBeerTitle(beer) }}
          </v-card-title>
          <v-card-title>{{ getBeerDisplayInformation(beer) }}</v-card-title>
          <v-card-text class="text--primary">
            <div class="text-md-h6">{{ getBeerStyle(beer) }}</div>
            <div class="text-md-body-1">
              Alcohol percentage: {{ getBeerAlcoholPercentage(beer) }}
            </div>
          </v-card-text>
        </v-card>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import { mapState } from 'vuex';

export default {
  name: 'Gallery',
  methods: {
    getBeerImage(beer) {
      return beer.images[0].image;
    },
    getBeerDisplayInformation(beer) {
      return beer.displayInformationPrice.price;
    },
    getBeerTitle(beer) {
      return beer.title;
    },
    getBeerStyle(beer) {
      return beer.style;
    },
    getBeerAlcoholPercentage(beer) {
      return beer.alcoholPercentage;
    },
  },
  computed: {
    ...mapState('menu', ['filteredAppData']),
  },
};
</script>

<style scoped>
  .text-ellipsis {
    text-overflow: ellipsis;

    /* Required for text-overflow to do anything */
    white-space: nowrap;
    overflow: hidden;
  }
</style>
