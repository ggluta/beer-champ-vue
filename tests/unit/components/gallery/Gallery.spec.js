import { shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import Vue from 'vue';
import Gallery from '../../../../src/components/gallery/Gallery.vue';

Vue.use(Vuex);

const mockBeer = {
  title: 'Brand Dubbelbock',
  images: [
    {
      image: 'https://www.beerwulf.com/globalassets/brand-dubbelbock.3_1_1.png',
      badge: {
        url: '/',
        alt: null,
        title: null,
      },
    },
  ],
  colors: ['yellow', 'malt', 'brown'],
  inStock: true,
  style: 'Bock Beer',
  alcoholPercentage: 7.5,
  volume: 30,
  displayInformationPrice: {
    currency: '£',
    price: '£ 2.39',
    filterPrice: 1.7925,
    discountPrice: '£ 1.79',
    discountPercentage: '-25',
    pricePerLiter: '£ 7.97',
  },
  containerType: 'Bottle',
};

const store = new Vuex.Store({
  modules: {
    menu: {
      namespaced: true,
      state: {
        appData: [mockBeer],
        filteredAppData: [mockBeer],
        title: mockBeer.title,
        style: mockBeer.style,
        color: mockBeer.colors[0],
        titleSelected: false,
        styleSelected: false,
        colorSelected: false,
      },
    },
  },
});

describe('Gallery test', () => {
  test('SHOULD render default WHEN normally instantiated', () => {
    const wrapper = shallowMount(Gallery, {
      store,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
