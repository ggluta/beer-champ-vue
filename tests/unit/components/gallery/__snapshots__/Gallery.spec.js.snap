// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery test SHOULD render default WHEN normally instantiated 1`] = `
<v-container id="gallery" fluid="">
  <v-layout row="" wrap="">
    <v-flex lg3="" md4="" sm6="" xs12="">
      <v-card class="ma-5 pa-lg-5">
        <v-img contain="" height="300px" src="https://www.beerwulf.com/globalassets/brand-dubbelbock.3_1_1.png"></v-img>
        <v-card-title class="text-md-h5 text-break">Brand Dubbelbock</v-card-title>
        <v-card-title>Â£ 2.39</v-card-title>
        <v-card-text class="text--primary">
          <div class="text-md-h6">Bock Beer</div>
          <div class="text-md-body-1">
            Alcohol percentage: 7.5
          </div>
        </v-card-text>
      </v-card>
    </v-flex>
  </v-layout>
</v-container>
`;
